forecasting/src/autogluon/forecasting/predictor.py

```diff
@@ -0,0 +1,420 @@
+            prediction_length=prediction_length,
+            val_data=val_data,
+            scheduler_options=scheduler_options,
+            hyperparameters=hyperparameters,
+            hyperparameter_tune=all(scheduler_options),
+            quantile_levels=quantile_levels,
+            time_limit=time_left,
+        )
+
+        self.save()
+        return self
+
+    # TODO: to be changed after ray tune integration
+    def _get_scheduler_options(
+        self,
+        hyperparameter_tune_kwargs: Optional[Union[str, Dict]],
+        time_limit: Optional[int] = None,
+    ) -> Tuple[Optional[Type], Optional[Dict[str, Any]]]:
+        """Validation logic for `hyperparameter_tune_kwargs`. Returns True if `hyperparameter_tune_kwargs` is None or
+        can construct a valid scheduler. Returns False if hyperparameter_tune_kwargs results in an invalid scheduler.
+        """
+        if hyperparameter_tune_kwargs is None:
+            return None, None
+
+        num_trials: Optional[int] = None
+        if isinstance(hyperparameter_tune_kwargs, dict):
+            num_trials = hyperparameter_tune_kwargs.get("num_trials")
+            if time_limit is None and num_trials is None:
+                logger.warning(
+                    "None of time_limit and num_trials are set, defaulting to num_trials=2",
+                )
+                num_trials = 2
+            else:
+                num_trials = hyperparameter_tune_kwargs.get("num_trials", 9999)
+        elif isinstance(hyperparameter_tune_kwargs, str):
+            num_trials = 9999
+
+        scheduler_cls, scheduler_params = scheduler_factory(
+            hyperparameter_tune_kwargs=hyperparameter_tune_kwargs,
+            time_out=time_limit,
+            nthreads_per_trial="auto",
+            ngpus_per_trial="auto",
+            num_trials=num_trials,
+        )
+
+        if scheduler_params["num_trials"] == 1:
+            logger.warning(
+                "Warning: Specified num_trials == 1 for hyperparameter tuning, disabling HPO. "
+            )
+            return None, None
+
+        scheduler_ngpus = scheduler_params["resource"].get("num_gpus", 0)
+        if (
+            scheduler_ngpus is not None
+            and isinstance(scheduler_ngpus, int)
+            and scheduler_ngpus > 1
+        ):
+            logger.warning(
+                f"Warning: ForecastingPredictor currently doesn't use >1 GPU per training run. "
+                f"Detected {scheduler_ngpus} GPUs."
+            )
+        return scheduler_cls, scheduler_params
+
+    def get_model_names(self) -> List[str]:
+        """Returns the list of model names trained in this `predictor` object."""
+        return self._trainer.get_model_names()
+
+    def predict(
+        self,
+        data: TimeSeriesDataFrame,
+        model: Optional[str] = None,
+        **kwargs,
+    ) -> TimeSeriesDataFrame:
+        """Return quantile and mean forecasts given a dataset to condition on.
```

Comment:
What if the user only wants the mean?

Reply:
Hmm, right now they "have" to want a quantile. Right now probabilistic forecasting is the "default" mode of operation for the library. The default validation score is mean quantile loss, and default `quantile_levels` is [0.1,...0.9]. In some sense we made probabilistic forecasting the "first class citizen" and point forecasting a side-effect of the library. I think this makes sense. Most libraries either choose one or the other. We prioritize quantile forecasting but still provide mean forecasts. If we want to design this back, I'm happy to discuss.

Reply:
Ok, we can keep for now